tests/test_helpers.py

```diff
@@ -234,18 +234,32 @@ def test_name_with_import_error(self, modules_tmp_path):
 
 
 class TestStreaming:
-    def test_streaming_with_context(self, app, client):
-        @app.route("/")
-        def index():
+    def test_stream_with_context_fails_with_async_route(self):
+        import gc
+
+        import pytest
+
+        import flask
+
+        app = flask.Flask(__name__)
+
+        @app.route("/stream")
+        async def stream():
+            @flask.stream_with_context
             def generate():
-                yield "Hello "
-                yield flask.request.args["name"]
-                yield "!"
+                yield "hello"
 
-            return flask.Response(flask.stream_with_context(generate()))
+            return flask.Response(generate())
 
-        rv = client.get("/?name=World")
-        assert rv.data == b"Hello World!"
+        with app.test_request_context("/stream"):
+            with pytest.raises(
+                RuntimeError, match="Install Flask with the 'async' extra"
+            ):
+                # This forces Flask to run the async view and raise the RuntimeError
+                app.ensure_sync(app.view_functions["stream"])()
```

Comment:
This is not testing the issue you linked to. This only raises when the async extra is not installed, and that's already tested elsewhere. This also fails when the async extra _is_ installed.